dependency.lic

```diff
@@ -1129,11 +1128,33 @@ class SetupFiles
       # Maps are treated as two-element arrays, first is the key, last is the value.
       # So we grab the last element which leaves us with map of spell properties.
       spell_data = spell_match.last
+      # Inject spell name into the spell properties.
+      # Some scripts check for either `abbrev` or `name`
+      # but our base spells data doesn't have name as a property, so we set it.
+      spell_data['name'] ||= spell_match.first
+
+      spell_data
     end
 
     # Find spell data by its abbreviation, case insensitive.
     get_spell_data_by_abbrev = lambda do |abbrev_to_find|
-      spells_data_values.find { |data| data['abbrev'].casecmp?(abbrev_to_find) }
+      # This creates a new map filtered to only the entries that match the spell abbreviation.
+      spell_matches = spells_data.select { |name, data| data['abbrev'].casecmp?(abbrev_to_find) }
```

Comment:
This method has been refactored to do exactly what `get_spell_data_by_name` does, but it's searching by abbreviation instead of by name.